mlflow/pytorch/torchserve/deploy/serve.py

```diff
@@ -0,0 +1,264 @@
+        if resp.status_code != 200:
+            raise Exception("Unable to list deployments")
+        return None
+
+    def update_deployment(self, name, model_uri=None, flavor=None, config=None):
+        """
+        Update the deployment with the name given at --name from the specified target
+
+        Using -C or --config additional parameters shall be updated for the corresponding model
+        """
+
+        query_path = ""
+
+        if config is not None:
+            for key in config:
+                query_path += "&" + key + "=" + str(config[key])
+
+            query_path = query_path[1:]
+
+        url = "{}/{}/{}?{}".format(self.management_api, "models", name, query_path)
+        resp = requests.put(url)
+
+        if resp.status_code != 202:
+            raise Exception("Unable to list deployments")
+        return {"flavor": flavor}
+
+    def list_deployments(self):
+        """
+        List the names of all model deployments in the specified target. These names can be used with
+        delete , update and get commands
+        """
+        url = "{}/{}".format(self.management_api, "models")
+        resp = requests.get(url)
+        if resp.status_code != 200:
+            raise Exception("Unable to list deployments")
+        return [resp.text]
+
+    def get_deployment(self, name):
+        """
+        Print the detailed description of the deployment with the name given at --name
+        in the specified target
+        """
+
+        url = "{}/{}/{}".format(self.management_api, "models", name)
+        resp = requests.get(url)
+        if resp.status_code != 200:
+            raise Exception("Unable to list deployments")
+        return {"deploy": resp.text}
+
+    def predict(self, deployment_name, input, output=None):
+        """
+        Predict using the inference api with the input file that is been specified using -I or --input
+        and outputs the result to either stdout or specified file
+        """
+
+        version = self.server_config["version"]
+        inp_path = str(input[0])
+        if os.path.exists(inp_path):
+            with open(inp_path, "r") as f:
+                given_input = json.loads(f.read())
+        else:
+            raise Exception("Input file not found")
+        url = "{}/{}/{}/{}".format(
+            self.inference_api, "predictions", deployment_name, version
+        )
+        resp = requests.post(url, given_input)
+        if resp.status_code != 200:
+            raise Exception("Unable to infer the results")
+
+        if output is not None and len(output) != 0:
+            output_path = str(output[0])
+            if os.path.exists(output_path):
+                with open(output_path + "/output.json", "w") as fp:
+                    json.dump({"result": resp.text}, fp)
+            else:
+                with open("output.json", "w") as fp:
+                    json.dump({"result": resp.text}, fp)
+        return resp.text
+
+    def generate_mar_file(
```

Comment:
You might want to make these private methods (prefix with `_`, like `_generate_mar_file`), unless you expect users to call these